### PR TITLE
Add queryLog for each tx action

### PR DESCRIPTION
**Pull request by cubismic**
This patch allows codechain-explorer to display the daily transaction log graph.

https://github.com/CodeChain-io/codechain-indexer/issues/191

### CODE REVIEW

File: src/worker/log.ts [hunk 1]
```diff
@@ -25,7 +26,77 @@ export async function indexLog(block: BlockAttribute, isRetracted: boolean) {
...
+                    case "unwrapCCC": {
+                        logtype = LogType.UNWRAP_CCC_COUNT;
+                        break;
+                    }
+                    default:
+                        logtype = undefined;
+                        break;
+                }
+                if (logtype !== undefined) {
+                    await queryLog(isRetracted, dateString, logtype, 1);
```
Comment (joojis) [inline src/worker/log.ts]: Why do you need to iterate this N times?
I think it's possible to aggregate the count for each type.

File: src/models/logic/transaction.ts [hunk 2]
```diff
@@ -536,6 +536,23 @@ export async function getTransactionsByAssetType(params: {
...
+                ...buildQueryForTransactions(params)
+            },
+            order: [["blockNumber", "DESC"], ["transactionIndex", "DESC"]],
+            include: fullIncludeArray
+        });
+    } catch (err) {
+        console.error(err);
+        throw Exception.DBError();
+    }
+}
```
Comment (joojis) [inline src/models/logic/transaction.ts]: This function is too general purpose to get the count the # of txs for each tx type.
I mean, you can just use `count` function with something like `group: ["type"]`

File: src/models/logic/transaction.ts [hunk 3]
```diff
@@ -536,6 +536,38 @@ export async function getTransactionsByAssetType(params: {
...
```
Comment (joojis) [inline src/models/logic/transaction.ts]: Please remove this function if you're not going to use it.

File: src/models/logic/transaction.ts [hunk 4]
```diff
@@ -536,6 +536,38 @@ export async function getTransactionsByAssetType(params: {
...
+
+export async function getNumberOfAggTransactionsByAssetType(params: {
```
Comment (joojis) [inline src/models/logic/transaction.ts]: getNumberOfEachTransactionType

File: src/worker/log.ts [hunk 5]
```diff
@@ -25,7 +26,86 @@ export async function indexLog(block: BlockAttribute, isRetracted: boolean) {
             parseInt(block.transactionsCount, 10)
         );
     }
-    // FIXME: queryLog for each tx action
+    getNumberOfAggTransactionsByAssetType({ blockNumber: block.number }).then(
```
Comment (joojis) [inline src/worker/log.ts]: return

File: src/worker/log.ts [hunk 6]
```diff
@@ -25,7 +26,86 @@ export async function indexLog(block: BlockAttribute, isRetracted: boolean) {
...
+        result => {
+            result.map(async resultTx => {
```
Comment (joojis) [inline src/worker/log.ts]: return Promise.all(...)

**Comment by cubismic** [comment]
I've changed my commit according to @joojis 's review.

**Pull request merged**